great_expectations/experimental/datasources/interfaces.py

```diff
@@ -346,9 +346,12 @@ class Datasource(
         "data_context_root_directory",  # filesystem argument
         "bucket",  # s3 argument
         "boto3_options",  # s3 argument
-        "prefix",  # s3 argument
-        "delimiter",  # s3 argument
+        "prefix",  # s3 argument and gcs argument
+        "delimiter",  # s3 argument and gcs argument
         "max_keys",  # s3 argument
+        "bucket_or_name",  # gcs argument
+        "gcs_options",  # gcs argument
+        "max_results",  # gcs argument
```

Comment:
`non-blocking`
I know we have a longer-term plan around this... but can't we also just extend and redefine this on each concrete `Datasource` class?

Reply:
@Kilo59 Let us work on this together in order to agree on the plan and the scope (and then we will document it and publish it).  Thank you.